trite-osprey/src/List.js

```diff
@@ -0,0 +1,219 @@
+        beerList = this.state.beerList
+
+    if (searchString.length > 0) {
+      beers = beers.filter(function(value) {
+        return value.beername.toLowerCase().match( searchString ) || value.brewery.toLowerCase().match( searchString )
+      })
+    }
+
+    const currentBeerDiv = this.state.currentBeer ? <div className={this.state.visible ? "popout" : "invisible"}>
+
+      <button onClick={this.hide.bind(this)} className="closebutton">X</button>
+
+      <div>{this.state.currentBeer.brewery}</div>
+
+      <div className='center_section'>
+
+          <div className='container_left'>
+            <img src={process.env.PUBLIC_URL + `/images/${this.state.currentBeer.image}.jpg`} style={{width: 180, height: 225}} alt="broken" />
+          </div>
+
+          <div className='container_right'>
+            <div> {this.state.currentBeer.beername} </div>
+            <div>{this.state.currentBeer.category}</div>
+            <div>ABU: { ' ' +this.state.currentBeer.abu}%</div>
+            <div> ${this.state.currentBeer.price}.00 </div>
+          </div>
+
+      </div>
+
+      <div className='description'>{this.state.currentBeer.description}</div>
+
+      <button className="addtocart" onClick={this.addToCart.bind(this)}>Add to Cart</button>
+      <button className="addtocart" onClick={this.removeFromCart.bind(this)}>Remove from Cart</button>
+
+    </div>
+    : null
+
+    const beerView = beers.map(function(value, index) {
+      return (
+      <div  onClick={ () => this.handleClick(event, index) } className="beerdisplay" >
+        <img key={index} src={process.env.PUBLIC_URL + `/images/${value.image}.jpg`} style={{width: 120, height: 180}} alt="broken" />
+      </div>
+      )
+    }, this)
+
+
+     return(
+
+        <div>
+{/*
+          <div className='cartContainer'>
+
+
+            <div className='price'>
+              <Cart className='beercartlist'/>
+              ${'' + this.state.totalPrice}
+
+              <ul onClick={this.addToCart.bind(this)} >
+                {beerList.map(function(value) {
+                  return (
+                    <div>
+                      {value}
+                    </div>
+                  )
+                })}
+              </ul>
+
+            </div>
+
+          </div> */}
+
+          <input
+            type="text"
+            value={this.state.searchString}
+            onChange={this.handleChange.bind(this)}
+            placeholder="Search beer"
+          />
+
+          <div className='wrapper'>
+            {beerView}
+          </div>
+
+          {currentBeerDiv}
+
+                    <div className='cartContainer'>
+
+
+                      <div className='price'>
+                        <Cart className='beercartlist'/>
+                        ${'' + this.state.totalPrice}
+
+                        <ul onClick={this.addToCart.bind(this)} >
+                          {beerList.map(function(value) {
```

Comment:
avoid mapping within JSX. Do it before your main JSX return section or make another component.